<launch>

	
	<param name="/auvsi16/video/right/resize/compressed/jpeg_quality" value="10" />
	<param name="/auvsi16/video/front/resize/compressed/jpeg_quality" value="10" />
	<node name="video_feed_upstream" pkg="auvsi16" type="video_feed_upstream" output="screen">
		<param name="resize_width_factor" value="0.3" />
		<param name="resize_height_factor" value="0.3" />
	</node>
	

	<node name="video_feed_front" pkg="auvsi16" type="video_feed_front" output="screen">
		<param name="camera_port" value="1" />
		<param name="camera_width" value="1280" />
		<param name="camera_height" value="720" />

		<param name="v4l2_configuration" value="0" />
	</node>

	<node name="video_feed_right" pkg="auvsi16" type="video_feed_right" output="screen">
		<param name="camera_port" value="0" />
		<param name="camera_width" value="1280" />
		<param name="camera_height" value="720" />

		<param name="v4l2_configuration" value="0" />
	</node>

	<!-- PID Here is deprecated change in the pid_controller  -->
	<!-- Change PID in the code by publishing to pid_const-->
	
	<arg name="pid_rate" default="100" />

	<arg name="kp" default="0" />
	<arg name="ki" default="0" />
	<arg name="kd" default="0" />
	<arg name="upper_limit" default="2000" />
	<arg name="lower_limit" default="-2000" />

	<!-- anti-windup is to clap integral value -->
	<arg name="anti_windup" default="0.1" />

	<arg name="topic_pid_in" default="auvsi16/pid/in" />
	<arg name="topic_pid_out" default="auvsi16/pid/out" />
	
	<node name="pid_controller" pkg="pid" output="screen" type="controller_adaptive" args="$(arg kp) $(arg ki) $(arg kd) $(arg pid_rate) -tfp $(arg topic_pid_in) -tfc $(arg topic_pid_out) -ul $(arg upper_limit) -ll $(arg lower_limit) -aw $(arg anti_windup)" />

	<node name="override_monitor" pkg="auvsi16" output="screen" type="override_monitor"/>
	<node name="motor_controller" pkg="auvsi16" output="screen" type="motor_controller"/>
	
	<node name="docking_mission_control" pkg="auvsi16" output="screen" type="docking_mission_control">
		<param name="heading_zero" value="49" />
		<param name="center_dock_lat" value="-6.367437" />
		<param name="center_dock_long" value="106.829882" />
	</node>

</launch>
